addons/loyalty/models/loyalty_reward.py

```diff
@@ -116,9 +117,13 @@ def _compute_multi_product(self):
     def _compute_description(self):
         for reward in self:
             reward_string = ""
-            if reward.reward_type == 'product':
+            if reward.description == _('Gift Card') or reward.description == _('eWallet'):
```

Comment:
It's best not to compare strings towards translated text, it will more than likely fail and not do what you want them to do.
We could hard-code it to gift card and ewallet depending on the program type instead.